lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -535,10 +871,136 @@ spec = describe "SHARED_TRANSACTIONS" $ do
         let walShared@(ApiSharedWallet (Right wal)) =
                 getFromResponse Prelude.id rPost
 
-        fundSharedWallet ctx faucetUtxoAmt walShared
+        fundSharedWallet ctx faucetUtxoAmt walShared Nothing Nothing
 
         return wal
 
+     fixtureTwoPartySharedWallet ctx = do
+
+        let index = 30
+        let passphrase = Passphrase $ BA.convert $ T.encodeUtf8 fixturePassphrase
+
+        -- first participant, cosigner 0
+        m15txtA <- liftIO $ genMnemonics M15
+        m12txtA <- liftIO $ genMnemonics M12
+        let (Right m15A) = mkSomeMnemonic @'[ 15 ] m15txtA
+        let (Right m12A) = mkSomeMnemonic @'[ 12 ] m12txtA
+        let accXPubDerivedA = accPubKeyFromMnemonics m15A (Just m12A) index passphrase
+
+        -- second participant, cosigner 1
+        m15txtB <- liftIO $ genMnemonics M15
+        m12txtB <- liftIO $ genMnemonics M12
+        let (Right m15B) = mkSomeMnemonic @'[ 15 ] m15txtB
+        let (Right m12B) = mkSomeMnemonic @'[ 12 ] m12txtB
+        let accXPubDerivedB = accPubKeyFromMnemonics m15B (Just m12B) index passphrase
```

Comment:
I'm reading through the tests and I think that more accurate flow from user perspective would be as follows. This is not demonstrated in the integration tests, but I think that realistically that how it would look like (correct me if I'm wrong):

**1. Creating wallets (Shared between two parties)**
Each party creates an "incomplete" wallet with the intention to "patch" it with other party's public key:
1st wallet owned by User A:
```
curl -X POST http://localhost:8090/v2/shared-wallets \
-d '{"mnemonic_sentence":["tuition","found","depth","salute","mix","client","end","door","deposit","false","guard","stairs","that","airport","sadness","harvest","change","enemy","finger","fly","squirrel","eagle","palace","parade"],
"passphrase":"Secure Passphrase 1st",
"name":"My Test Shared Wallet 1st",
"account_index":"0H",
"payment_script_template":{"cosigners":{"cosigner#0":"self"},"template":{"all":["cosigner#0","cosigner#1"]}}}' \
-H "Content-Type: application/json"
```
2nd wallet owned by User B:
```
curl -X POST http://localhost:8090/v2/shared-wallets \
-d '{"mnemonic_sentence":["grid","glare","blur","interest","actress","wheel","afraid","satisfy","jump","hard","regular","choice","era","approve","nice","banner","rubber","tuna","champion","volcano","expect","unfair","ability","expose"],
"passphrase":"Secure Passphrase 2nd",
"name":"My Test Shared Wallet 2nd",
"account_index":"0H",
"payment_script_template":{"cosigners":{"cosigner#0":"self"},"template":{"all":["cosigner#0","cosigner#1"]}}}' \
-H "Content-Type: application/json"
```

**2. Patching wallet**
In order to be able to basically be able to use wallet i.e. derive addresses, sign or submit transactions both parties need to have "complete" wallets with public keys from the other party patched to their own wallet. They need to get their account public key from their "incomplete" wallet and share it with other party. For getting public key they can use [`GET /shared-wallets/{walletId}/keys`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/getAccountKeyShared).

User A:
* note the key needs to be re-formatted from bech32 to hex format
```
curl -X GET http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/keys?format=extended
acct_shared_xvk16kf4l5xaul4xntd3kr5l0yeuv3s62hzwzdjm7ckv742fzkna6dapqcy42a9a602xld2pg8rtzwkv5gyue5faaj8ljap6tsnj2n4t5jg82uvmc

bech32 <<< acct_shared_xvk16kf4l5xaul4xntd3kr5l0yeuv3s62hzwzdjm7ckv742fzkna6dapqcy42a9a602xld2pg8rtzwkv5gyue5faaj8ljap6tsnj2n4t5jg82uvmc

d5935fd0dde7ea69adb1b0e9f7933c6461a55c4e1365bf62ccf554915a7dd37a106095574bdd3d46fb54141c6b13acca209ccd13dec8ff9743a5c27254eaba49
```

User B:
```
curl -X GET http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/keys?format=extended
acct_shared_xvk18td43rutqvayyakxs2r0qx5e9jyzrrajz7rl46h6wvfcqsn2yv63p6llv4umlz6frqu69knlt6sd7shtt8c4l9h9jharqjllgeeq33czat7xa

bech32 <<< acct_shared_xvk18td43rutqvayyakxs2r0qx5e9jyzrrajz7rl46h6wvfcqsn2yv63p6llv4umlz6frqu69knlt6sd7shtt8c4l9h9jharqjllgeeq33czat7xa

3adb588f8b033a4276c68286f01a992c88218fb21787faeafa731380426a233510ebff6579bf8b491839a2da7f5ea0df42eb59f15f96e595fa304bff467208c7
```
User A patches his wallet with User's B public key:
```
curl -X PATCH http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/payment-script-template \
-d '{"cosigner#1":"3adb588f8b033a4276c68286f01a992c88218fb21787faeafa731380426a233510ebff6579bf8b491839a2da7f5ea0df42eb59f15f96e595fa304bff467208c7"}' \
-H "Content-Type: application/json"
```
User B patches his wallet with User's A public key:
```
curl -X PATCH http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/payment-script-template \
-d '{"cosigner#1":"d5935fd0dde7ea69adb1b0e9f7933c6461a55c4e1365bf62ccf554915a7dd37a106095574bdd3d46fb54141c6b13acca209ccd13dec8ff9743a5c27254eaba49"}' \
-H "Content-Type: application/json"
```

**3. Transaction**
User A creates very basic transaction posting simple metadata:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-construct \
-d '{"metadata":{"1":14}}' \
-H "Content-Type: application/json" | jq .transaction

hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTbGAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56hAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO
```
User A signs this transaction:

```
$ curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-sign \
> -d '{"passphrase":"Secure Passphrase","transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56hAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO"}' \
> -H "Content-Type: application/json" | jq

{
  "transaction": "hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhA+Rvr2bWwGTp+hgDwu59mZYdlLCuCmtLbAd8wjbsAFk+q1Sks5y81MF5+RqQ/L5w5kp1yDX3GggrFG2J3bXBeBAGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="
}
```
At this point User A can decode his signed tx to see if his witness has been included:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-decode -d '{"transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhA+Rvr2bWwGTp+hgDwu59mZYdlLCuCmtLbAd8wjbsAFk+q1Sks5y81MF5+RqQ/L5w5kp1yDX3GggrFG2J3bXBeBAGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' -H "Content-Type: application/json" | jq .witness_count

{
  "bootstrap": 0,
  "scripts": [
    {
      "script": {
        "all": [
          "policy_vkh1ca5u96qkrknyn7n408ckj2g9fmf3tvkl4274mxsgpx6uu8ry5al",
          "policy_vkh1ut3p57zhx5xc3mk7cwf8yx5md7zlg5yvexhjj4hy3usku9l2mv0"
        ]
      },
      "script_type": "native"
    }
  ],
  "verification_key": 1
}
```
`"verification_key": 1` so I believe it is as expected.
**Question:** what is the difference between `bootstrap` and `verification_key`... explanation for both in swagger is: The number of witnesses detected

Now User A needs to send CBOR of signed tx to User B and ask him to sign it with his keys, and user B can use his wallet to do so:
User B signs transaction provided by User A:

```
curl -X POST http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/transactions-sign \
> -d '{"passphrase":"Secure Passphrase 2nd","transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhA+Rvr2bWwGTp+hgDwu59mZYdlLCuCmtLbAd8wjbsAFk+q1Sks5y81MF5+RqQ/L5w5kp1yDX3GggrFG2J3bXBeBAGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' \
> -H "Content-Type: application/json" | jq

{
  "transaction": "hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhA+Rvr2bWwGTp+hgDwu59mZYdlLCuCmtLbAd8wjbsAFk+q1Sks5y81MF5+RqQ/L5w5kp1yDX3GggrFG2J3bXBeBAGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="
}

```
User B decodes that he signed to see if witness was added:
```
curl -X POST http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/transactions-decode \
> -d '{"transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhA+Rvr2bWwGTp+hgDwu59mZYdlLCuCmtLbAd8wjbsAFk+q1Sks5y81MF5+RqQ/L5w5kp1yDX3GggrFG2J3bXBeBAGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' \
> -H "Content-Type: application/json" | jq .witness_count

{
  "bootstrap": 0,
  "scripts": [
    {
      "script": {
        "all": [
          "policy_vkh1ca5u96qkrknyn7n408ckj2g9fmf3tvkl4274mxsgpx6uu8ry5al",
          "policy_vkh1ut3p57zhx5xc3mk7cwf8yx5md7zlg5yvexhjj4hy3usku9l2mv0"
        ]
      },
      "script_type": "native"
    }
  ],
  "verification_key": 1
}
```
:warning: `"verification_key": 1`... since User B "signed" this transaction it should be 2, right?

Ok, now User B sends back signed CBOR to User A such that he can submit the transaction...
User A tries to submit:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-submit \
-d '{"transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTarAgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhA+Rvr2bWwGTp+hgDwu59mZYdlLCuCmtLbAd8wjbsAFk+q1Sks5y81MF5+RqQ/L5w5kp1yDX3GggrFG2J3bXBeBAGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' \
-H "Content-Type: application/json" | jq

{
  "code": "missing_witnesses_in_transaction",
  "message": "The transaction expects 2 witness(es) to be fully-signed but 1 was provided. Submit fully-signed transaction."
}
```
Seems that witness that User B wanted to include was not included. Am I missing something here?
Is this workflow, in general, correct?

Reply:
1. there are three types of wintesses in witness set in ledger/node : bootstrap, from verification key and script structure ones. When it comes to the first two, at this moment, I only log the count of them. When it comes to script of key hashes I enumerate them.
2. wintess from cosigner0 and cosigner1 having unique and different mnemonics should be different and hence should be addable upon signing. It seems, for the reasons I do not understand, that cosigner1 adds THE SAME witness as cosigner0. Hence the problems

Reply:
Ok, changing previous flow slightly then.
User B creates wallet:
```
curl -X POST http://localhost:8090/v2/shared-wallets \
-d '{"mnemonic_sentence":["grid","glare","blur","interest","actress","wheel","afraid","satisfy","jump","hard","regular","choice","era","approve","nice","banner","rubber","tuna","champion","volcano","expect","unfair","ability","expose"],
"passphrase":"Secure Passphrase 2nd",
"name":"My Test Shared Wallet 2nd",
"account_index":"0H",
"payment_script_template":{"cosigners":{"cosigner#1":"self"},"template":{"all":["cosigner#0","cosigner#1"]}}}' \
-H "Content-Type: application/json"
```
(now user B is explicitely 'cosigner#1')

User B patches wallet with User A's pub key:
```
curl -X PATCH http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/payment-script-template \
-d '{"cosigner#0":"d5935fd0dde7ea69adb1b0e9f7933c6461a55c4e1365bf62ccf554915a7dd37a106095574bdd3d46fb54141c6b13acca209ccd13dec8ff9743a5c27254eaba49"}' \
-H "Content-Type: application/json"
```
(User B is patching User A as 'cosigner#0')

**Now making tx again:**
User A constructs:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-construct \
-d '{"metadata":{"1":14}}' \
-H "Content-Type: application/json" | jq .transaction

hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56hAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO
```

User A signs:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-sign \
> -d '{"passphrase":"Secure Passphrase","transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56hAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO"}' \
> -H "Content-Type: application/json" | jq

{
  "transaction": "hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDQGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="
}

```

User B signs:
```
curl -X POST http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/transactions-sign \
> -d '{"passphrase":"Secure Passphrase 2nd","transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDQGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' \
> -H "Content-Type: application/json" | jq

{
  "transaction": "hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDQGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="
}

```

User A submits:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-submit \
-d '{"transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDQGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' \
-H "Content-Type: application/json"

{"code":"missing_witnesses_in_transaction",
 "message":"The transaction expects 2 witness(es) to be fully-signed but 1 was provided. Submit fully-signed transaction."}
```

Reply:
Ok, now creating User B wallet in one go (without patching):
```
curl -X POST http://localhost:8090/v2/shared-wallets \
-d '{"mnemonic_sentence":["grid","glare","blur","interest","actress","wheel","afraid","satisfy","jump","hard","regular","choice","era","approve","nice","banner","rubber","tuna","champion","volcano","expect","unfair","ability","expose"],"passphrase":"Secure Passphrase 2nd","name":"My Test Shared Wallet 2nd","account_index":"0H","payment_script_template":{"cosigners":{"cosigner#0":"d5935fd0dde7ea69adb1b0e9f7933c6461a55c4e1365bf62ccf554915a7dd37a106095574bdd3d46fb54141c6b13acca209ccd13dec8ff9743a5c27254eaba49","cosigner#1":"self"},"template":{"all":["cosigner#0","cosigner#1"]}}}' \
-H "Content-Type: application/json"
```
again User A is cosigner#0, user B is cosigner#1.

 - user A constructs and signs tx
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-construct \
> -d '{"metadata":{"1":14}}' \
> -H "Content-Type: application/json" | jq .transaction

"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTzvwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56hAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO"

curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-sign \
> -d '{"passphrase":"Secure Passphrase","transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56hAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO"}' \
> -H "Content-Type: application/json" | jq

{
  "transaction": "hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDQGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="
}

```

User B signs:
```
curl -X POST http://localhost:8090/v2/shared-wallets/4e04d1684767aeba3ed289f163bb0b34b34b49f8/transactions-sign \
 -d '{"passphrase":"Secure Passphrase 2nd","transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIGCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDQGBggGCggBYHMdpwugWHaZJ+nV58WkpBU7TFbLfqr1dmggJtc6CAFgc4uIaeFc1DYju3sOSchqbb4X0UIzJrylW5I8hbvXZAQOhAKEBDg=="}' \
 -H "Content-Type: application/json" | jq

{
  "transaction": "hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIKCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDYJYIMcnW8oziNy+Vf1IjIe+IcJMd8kGIplZJK9rQX/Tt3EMWEDhozs+XEitdBe1JkzXX3q8tKI2HfLul98mp0V0a4OQYIjHjrDBImLN6O3lf26O3cugPoluxVneKbSuOM/qZQMFAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO"
}
```
User A submits:
```
curl -X POST http://localhost:8090/v2/shared-wallets/8954aaaa3357ed1912759371f9657934f0955c17/transactions-submit \
-d '{"transaction":"hKYAgYJYILQohoEnFMxr2d0lF1iu4B3nwmtuF2/mBQHp6UJfI99MAAGBogBYHXCdtpPKXcep7Fze+5UXp1HTR6dFNUtQZEZFdNnGARsAAAACVAk1zwIaAAKuMQMaAGTvFwgAB1ggftmsnMQXAnuzG7Aj+wG1Rg0wCA4KQ7icOoJG0p95+56iAIKCWCB/57/l4rZzRaJE2YFl02Pjjv5gGQ0MOK/fgcMpDXIrKlhAYakw+drZVn4M0wVDUBJdLM+Rfk68YC4tdFzLGGXbwIOe1KsDi5Zddwxk5dy2VEWqe0dPURZJuMaTmQ/uo2diDYJYIMcnW8oziNy+Vf1IjIe+IcJMd8kGIplZJK9rQX/Tt3EMWEDhozs+XEitdBe1JkzXX3q8tKI2HfLul98mp0V0a4OQYIjHjrDBImLN6O3lf26O3cugPoluxVneKbSuOM/qZQMFAYGCAYKCAFgcx2nC6BYdpkn6dXnxaSkFTtMVst+qvV2aCAm1zoIAWBzi4hp4VzUNiO7ew5JyGptvhfRQjMmvKVbkjyFu9dkBA6EAoQEO"}' \
-H "Content-Type: application/json"

{"id":"90f2aa143f55ca4fc88fec93b5ea1ef1e44f866d256f90f8e8705de61ffd5c46"}
```
Ok, it works fine when PATCHing is not involved on User B side.

Reply:
Thanks Piotr for checking hypothesis. Let's move on with current way of instantiating active wallet in this PR. And solve patching as a separate ticket and PR.

Reply:
I've created two separate tickets:
 - ADP-2331 - addressing the PATCHing issue
 - ADP-2332 - addressing the fact that delegation/payment templates don't support `bech32` format (https://github.com/input-output-hk/cardano-wallet/pull/3529#discussion_r1003066748)

Reply:
thanks Piotr - I will tackle after this PR is merged